x-pack/plugins/endpoint/public/applications/endpoint/store/hosts/host_pagination.test.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreStart } from 'kibana/public';
+import { DepsStartMock, depsStartMock } from '../../mocks';
+import { AppAction, HostState, HostIndexUIQueryParams } from '../../types';
+import { Immutable } from '../../../../../common/types';
+import { History, createBrowserHistory } from 'history';
+import { hostMiddlewareFactory } from './middleware';
+import { applyMiddleware, Store, createStore } from 'redux';
+import { hostListReducer } from './reducer';
+import { coreMock } from 'src/core/public/mocks';
+import { urlFromQueryParams } from '../../view/hosts/url_from_query_params';
+import { uiQueryParams } from './selectors';
+
+describe('host list pagination: ', () => {
+  let store: Store<Immutable<HostState>, Immutable<AppAction>>;
+  let fakeCoreStart: jest.Mocked<CoreStart>;
+  let depsStart: DepsStartMock;
+  let history: History<never>;
+  let queryParams: () => HostIndexUIQueryParams;
+
+  let historyPush: (params: HostIndexUIQueryParams) => void;
+  beforeEach(() => {
+    fakeCoreStart = coreMock.createStart();
+    depsStart = depsStartMock();
+    history = createBrowserHistory();
+
+    const middleware = hostMiddlewareFactory(fakeCoreStart, depsStart);
+    store = createStore(hostListReducer, applyMiddleware(middleware));
+
+    history.listen(location => {
+      store.dispatch({ type: 'userChangedUrl', payload: location });
+    });
+
+    queryParams = () => uiQueryParams(store.getState());
+
+    historyPush = (nextQueryParams: HostIndexUIQueryParams): void => {
+      return history.push(urlFromQueryParams(nextQueryParams));
+    };
+  });
+
+  describe('when a new page size is passed', () => {
+    beforeEach(() => {
+      historyPush({ ...queryParams(), page_size: '20' });
+    });
+    it('should modify the url correctly', () => {
+      expect(queryParams()).toMatchInlineSnapshot(`
+        Object {
+          "page_index": "0",
+          "page_size": "20",
+        }
+      `);
+    });
+  });
+  describe('when an invalid page size is passed', () => {
```

Comment:
Are there any other variations you should be testing? Example: what if multiple `page_size` params are in the URL? what if `page_size` is `-1`? or a string (`jsdkfjkds`)?

Reply:
Yes, I agree with @paul-tavares  - @parkiino we should test the following invalid values as well:
- -1
- invalid UUID 
- valid UUID + special characters
- valid UUID + JS embed
- valid UUID + unicode
- empty string 

If it's too much of a lift, I can also test these examples manually and create another ticket to automate these tests in the future. I will also manually test when page_index is more than what we expect to make sure it results in the same bug I found in the policy list URL pagination